5_evaluate.R

```diff
@@ -8,54 +8,51 @@ p5 <- list(
-  # then we could use that first target here to get NLDAS in long format, per https://github.com/USGS-R/mntoha-data-release/blob/main/src/eval_utils.R#L23-L30
-  tar_target(
-    p5_nldas_preds_long,
-    munge_long(p3_nldas_glm_uncalibrated_output) %>%
-      mutate(site_id = p2_nldas_glm_uncalibrated_run_groups$site_id, .before=1) %>%
-      select(-ice),
-    pattern = map(p3_nldas_glm_uncalibrated_output, p2_nldas_glm_uncalibrated_run_groups)
-  ),
+  # Get vector of site_ids for which we have NLDAS output
+  tar_target(p3_nldas_export_site_ids,
+             p2_nldas_glm_uncalibrated_run_groups %>%
+               pull(site_id)),
   
   # Prep site observations
-  ### Bring in `7b_temp_merge/out/merged_temp_data_daily.feather` from `lake-temperature-model-prep`
-  ### (would be target in 1_prep.R)
-  # filter obs to sites for which we have NLDAS output
-  ### Use https://github.com/USGS-R/mntoha-data-release/blob/main/src/file_utils.R#L441-L446
-  ### Passing `driver_start_date` and `driver_end_date` from `p1_nldas_dates`
-  # And subset obs to those for sites w/ >= 10 dates with observations
-  ### Use https://github.com/USGS-R/mntoha-data-release/blob/main/src/eval_utils.R#L4-L11
+  # filter obs to sites and dates for which we have NLDAS output
+  # And further filter obs to those for sites w/ >= 10 dates with observations
   tar_target(p5_obs_for_eval,
-             get_eval_obs(p1_obs_temps, min_dates = 10)),
+             get_eval_obs(p1_obs_feather, p3_nldas_export_site_ids, p1_nldas_dates$driver_start_date, 
+                          p1_nldas_dates$driver_end_date, min_obs_dates = 10)),
   
-  # Get vector of evaluation sites
+  # Get vector of evaluation sites, based on availability of observations
   tar_target(p5_eval_sites,
              p5_obs_for_eval %>% pull(site_id) %>% unique()),
-  
-  # Group subsetted obs by site, set up tar_group()
+
+  # Group filtered obs by site, set up tar_group()
   tar_target(p5_obs_for_eval_groups,
              p5_obs_for_eval %>%
                group_by(site_id) %>%
                tar_group(),
              iteration = "group"),
   
-  # filter long-format NLDAS preds to only evaluation sites
+  # Prep NLDAS predictions
+  # Read in predictions for sites that are evaluation sites and filter
+  # predictions for each site to only those dates that have observations 
   tar_target(
     p5_nldas_preds_eval,
-    p5_nldas_preds_long %>%
-      filter(site_id %in% p5_eval_sites)
+    {
+      tar_assert_identical(p5_eval_sites, unique(p5_obs_for_eval_groups$site_id), 
+                           "p5_eval_sites site id doesn't match p5_obs_for_eval_groups site id")
+      arrow::read_feather(sprintf('3_extract/out/GLM_%s_NLDAS.feather', p5_eval_sites)) %>%
+        filter(time %in% p5_obs_for_eval_groups$time) %>%
+        mutate(site_id = p5_eval_sites) %>%
+        select(-ice) %>%
+        arrange(site_id, time)
+    },
+    pattern = map(p5_eval_sites, p5_obs_for_eval_groups)
```

Comment:
I think splitting this & the steps in `match_pred_obs()` is not necessary and is making the mapping more complicated. I think that you can actually join those together and skip some of these intermediate targets. After `p5_obs_for_eval_groups` and before `p5_nldas_pred_obs_csv`, you really only need one target (move some code from the target makefile into `match_pred_obs()` instead. It simplifies the mapping and eliminates any concern that we might not be using the same sites. 

```r
tar_target(p5_nldas_pred_obs, 
        match_pred_obs(
          file_in = sprintf("3_extract/out/GLM_%s_NLDAS.feather", unique(p5_obs_for_eval_groups$site_id)),
          site_obs_data = p5_obs_for_eval_groups
        ),
      pattern = map(p5_obs_for_eval_groups)
      )
    )
```

and `match_pred_obs()` would look like:

```r
match_pred_obs <- function(file_in, eval_obs) {
  eval_preds <- arrow::read_feather(file_in) %>%
    filter(time %in% eval_obs$time) %>%
    mutate(site_id = unique(eval_obs$site_id)) %>%
    select(-ice) %>%
    arrange(site_id, time)
  # The rest of what is already in `match_pred_obs()` goes here
}
```

You could go so far as to include the step to write out as a CSV within this step & then just have the next step `prep_data_for_eval()` read in a CSV.

Reply:
Sounds good -- agreed that that would clean it up and removes risk re: not using the same sites. I'll make those edits now

Reply:
Okay edits pushed. I think I landed in that awkward setup of targets after gradually shifting away from using the already read in predictions. Definitely seems cleaner to have this reduced # of targets now that we're simply reading in the predictions for each site.